microsoft365defender_connector.py

```diff
@@ -781,6 +781,34 @@ def _remove_tokens(self, action_result):
             if self._state.get(DEFENDER_CODE_STRING):
                 self._state.pop(DEFENDER_CODE_STRING)
 
+    def _fix_up_odata_fields(self, response):
+        """Fields containing a period are incompatible with data path syntax. Create fields that are accessible instead
+
+        :param response: Dictionary containing the response by the Defender API
+        :return: Dictionary containing the response by the Defender API with additional fields.
+        """
+
+        if not isinstance(response, dict):
+            return response
+
+        # @odata.context
```

Comment:
This is ok, but in the future it would be great if this could be more dynamic. Seems like you're taking out the `@` for the key value. Having a list of values that qualify for this adjustment would be along the lines of what would be better.

Reply:
Sure. We had looked for values with a similar format but these values seem to be the most used ones and the only ones appearing in responses for most of the actions apart from run query where the response largely varies on customer query and data. Even in that the values covered in this solution seem to be of most importance with relevant information, and hence we went ahead with this approach.
Thanks.